src/plugins/intel_cpu/src/nodes/concat.cpp

```diff
@@ -58,6 +58,9 @@ Concat::Concat(const std::shared_ptr<ngraph::Node>& op, const dnnl::engine& eng,
     }
 
     const auto inRank = getInputShapeAtPort(0).getRank();
+    if (inRank > 6) {
```

Comment:
I get your point. I miss that the oneDNN have two concat primitives. The `simple_concat` supports rank < 6 while `ref_concat` handles other cases. Therefore, the correct order should be 
1. Check if rank < 6, if so, then run our internal concat
2. Otherwise, we use onednn version of ref_concat.